examples/cli-version-compare/run.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -e
+
+LDIR="$(find . ! -name . -prune -type d -exec sh -c 'echo "$1" | wc -m' shell {} \; | sort -h | tail -n 1)"
+ROOT="$( ([ "$REACH_DOCKER" = 0 ] && pwd) || echo /app/src)"
+
+e () {
+  printf '\e[34;1mRunning %s...\e[0m' "$2"
+  if [ "$REACH_DOCKER" = 0 ]; then
+    echo
+  fi
+
+  set +e
+  o=$(REACH_VERSION="$( ([ -f "./$2/REACH_VERSION" ] && cat "./$2/REACH_VERSION") || echo)" \
+    ../../reach version-compare2 \
+      --non-interactive \
+      --ils="$ROOT/$2/l.json" \
+      --stub-remote="$ROOT/$2/r.json" \
+      --stub-script="./$2/reach") # skip $ROOT because `diff` is evaluated outside of Docker
+  c=$?
+  set -e
+
+  e="$(cat "./$2/o.txt")"
+  if [ ! $c -eq "$1" ] || [ ! "$o" = "$e" ]; then
+    echo
+
+    if [ ! $c -eq "$1" ]; then
+      printf '\e[31;1mExpected %s to exit with %s but got %s.\e[0m\n' "$2" "$1" "$c"
+    fi
+
+    if [ ! "$o" = "$e" ]; then
+      printf '\e[31;1mUnexpected output:\e[0m\n'
+      echo "$o" | diff --color "./$2/o.txt" -
+    fi
+
+    echo
+    exit 1
+  fi
+
+  if [ "$REACH_DOCKER" = 0 ]; then
+    printf '\e[1;32m ✔\e[0m Exited with %s\n' "$1"
+  else
+    printf '\e[1;32m  %*s\e[0m Exited with %s' "$((LDIR - ${#2}))" "✔" "$1"
+  fi
+  echo
+}
+
+e  1 unknown-img
+e  1 unknown-tag:0.1.7
+e  1 unknown-tag:dab71275
+e 60 new-images:reach-cli+react-runner+runner
+e 60 new-images:rpc-server#new-tags:reach-cli+devnet-cfx
+e 60 new-tags:reach+react-runner
+e  0 fully-synchronized
+e  0 connector-avail:eth+cfx
+e  0 connector-avail-3rd-party:image-not-present
+e 60 script#new-images:devnet-algo+postgres
+e 60 script#fully-synchronized
+e  0 connector-avail-3rd-party:image-present-tag-not
+e  0 connector-avail-3rd-party:image-present-sha-not
```

Comment:
I would prefer something that looks at the every subdirectory and gets the expected error code from something in the directory. I worry about someone thinking they added a test case (by putting in a subdir) but forgetting to add it here

Reply:
I found this more convenient when focusing on a single test and less scattered but I'll make a separate PR if you think that's better.

Reply:
I've created CORE-1134 to track it.